lib/plugsnag/error_report_builder.ex

```diff
@@ -1,4 +1,7 @@
 defmodule Plugsnag.ErrorReportBuilder do
 
   @callback build_error_report(Plugsnag.ErrorReport.t, Plug.Conn.t) :: Plugsnag.ErrorReport.t
+  @callback build_error_report(Plugsnag.ErrorReport.t, Plug.Conn.t, Exception.t) :: Plugsnag.ErrorReport.t
+
+  @optional_callbacks build_error_report: 2, build_error_report: 3
```

Comment:
I don't really like this but I couldn't figure out a way to keep backwards compatibility and suppress warnings.